sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -1392,3 +1394,17 @@ object UpdateNullabilityInAttributeReferences extends Rule[LogicalPlan] {
       }
   }
 }
+
+/**
+ * Set the seed for random number generation in Uuid expressions for streaming query.
+ */
+object ResolvedUuidExpressionsForStreaming extends Rule[LogicalPlan] {
+  private lazy val random = new Random()
+
+  override def apply(plan: LogicalPlan): LogicalPlan = plan.transformUp {
+    case p => p transformExpressionsUp {
+      case Uuid(_) if p.isStreaming => Uuid(Some(random.nextLong()))
```

Comment:
not a big deal at all but can we remove `(_)` like _: Uuid ?

Reply:
Yeah, sure.